fl4health/clients/moon_client.py

```diff
@@ -89,14 +91,17 @@ def get_contrastive_loss(
 
         return self.ce_criterion(logits, labels)
 
-    def set_parameters(self, parameters: NDArrays, config: Config) -> None:
+    def get_parameters(self, config: Config) -> NDArrays:
         assert isinstance(self.model, MoonModel)
-        # Save the parameters of the old local model
+        # Save the parameters of the old LOCAL model
```

Comment:
Why is the saving being done on get now instead of set?

Reply:
That's a good question and the crux of the issue that Sana identified. `set_parameters` is called to start both the `fit` and `eval` stages. The eval stage happens after aggregation. So in the simplified case, when `old_models_list` is supposed to have size 1, the `old_local_model` that MOON needs to compute features is replaced with the global model produced by aggregation. `get_parameters` is only called at the end of each `fit`, which means that old_models_list will always have the most recent model produced by **local** training, which is what we want.

Reply:
Just for the record, I think this would be better if moved to its own function, but as per offline conversation, this will be done in a follow up ticket (https://app.clickup.com/t/86879vn75)